implants/golem/src/main.rs

```diff
@@ -16,25 +17,35 @@ struct ParsedTome {
 
 async fn run_tomes(tomes: Vec<ParsedTome>) -> Result<Vec<String>> {
     let mut runtimes = Vec::new();
+    let mut idx = 1;
     for tome in tomes {
-        let runtime = eldritch::start(Tome {
-            eldritch: tome.eldritch,
-            parameters: HashMap::new(),
-            file_names: Vec::new(),
-        })
+        let runtime = eldritch::start(
+            idx,
+            Tome {
+                eldritch: tome.eldritch,
+                parameters: HashMap::new(),
+                file_names: Vec::new(),
+            },
+        )
         .await;
         runtimes.push(runtime);
+        idx += 1;
     }
 
     let mut result = Vec::new();
     for runtime in &mut runtimes {
         runtime.finish().await;
-        let mut out = runtime.collect_text();
-        let errors = runtime.collect_errors();
-        if !errors.is_empty() {
-            return Err(anyhow!("tome execution failed: {:?}", errors));
+
+        let messages = runtime.collect();
+        for msg in messages {
+            match msg {
```

Comment:
Is reporting other types going to be managed here as well?

Reply:
yeee. ALL reporting types